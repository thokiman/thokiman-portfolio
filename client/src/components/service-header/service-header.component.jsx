import React from "react";
import { withRouter } from "react-router-dom";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import { CSSTransitionGroup } from "react-transition-group";

import {
  selectArtService,
  selectEngineerService,
  selectFullstackService,
  selectPhotoService,
} from "../../redux/service/service.selectors";
import {
  LeftServiceContainer,
  DirectoryService,
  DirectoryServiceTextEng,
  DirectoryServiceTextFsd,
  DirectoryServiceTextGd,
  DirectoryServiceTextPg,
} from "./service-header.styles";
import { selectIsSideBarHidden } from "../../redux/header/header.selectors";

const ServiceHeader = ({
  location: { pathname },
  art,
  photo,
  engineer,
  fullstack,
  isSideBarHidden,
}) => {
  return (
    <LeftServiceContainer $issidebarhidden={isSideBarHidden}>
      <CSSTransitionGroup
        transitionAppear={true}
        transitionAppearTimeout={800}
        transitionEnterTimeout={800}
        transitionLeaveTimeout={200}
        transitionName={"slide-out"}
      >
        <DirectoryService>
          <DirectoryServiceTextEng
            to={engineer.routeName}
            $matchpath={!!pathname.match(RegExp(`${engineer.routeName}$`))}
          >
            engineer
          </DirectoryServiceTextEng>
          <DirectoryServiceTextFsd
            to={fullstack.routeName}
            $matchpath={!!pathname.match(RegExp(`${fullstack.routeName}$`))}
          >
            full
            <br />
            stack <br />
            developer
          </DirectoryServiceTextFsd>
          <DirectoryServiceTextGd
            to={art.routeName}
            $matchpath={!!pathname.match(RegExp(`${art.routeName}$`))}
          >
            graphic
            <br />
            design
          </DirectoryServiceTextGd>
          <DirectoryServiceTextPg
            to={photo.routeName}
            $matchpath={!!pathname.match(RegExp(`${photo.routeName}$`))}
          >
            photo
            <br />
            graphy
          </DirectoryServiceTextPg>
        </DirectoryService>
      </CSSTransitionGroup>
    </LeftServiceContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  art: selectArtService,
  photo: selectPhotoService,
  engineer: selectEngineerService,
  fullstack: selectFullstackService,
  isSideBarHidden: selectIsSideBarHidden,
});

export default withRouter(connect(mapStateToProps)(ServiceHeader));
