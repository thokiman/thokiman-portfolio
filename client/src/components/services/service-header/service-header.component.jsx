import React from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { createStructuredSelector } from 'reselect';
import { selectIsSideBarActive } from '../../../redux/header/header.selectors';
import {
  selectArtService,
  selectDevService,
  selectEngineerService,
  selectPhotoService,
} from '../../../redux/service/service.selectors';
import {
  DirectoryService,
  DirectoryServiceTextEng,
  DirectoryServiceTextFsd,
  DirectoryServiceTextGd,
  DirectoryServiceTextPg,
  LeftServiceContainer,
} from './service-header.styles';

export const ServiceHeader = ({
  location: { pathname },
  art,
  photo,
  engineer,
  dev,
  isSideBarActive,
}) => {
  return (
    <LeftServiceContainer
      className='left-service-container'
      $issidebaractive={isSideBarActive}
    >
      <DirectoryService>
        <DirectoryServiceTextEng
          className='directory-service-header'
          to={engineer.routeName}
          $matchpath={!!pathname.match(RegExp(`${engineer.routeName}$`))}
        >
          chemical
          <br />
          engineer
        </DirectoryServiceTextEng>
        <DirectoryServiceTextFsd
          className='directory-service-header'
          to={dev.routeName}
          $matchpath={!!pathname.match(RegExp(`${dev.routeName}$`))}
        >
          full
          <br />
          stack
          <br />
          developer
        </DirectoryServiceTextFsd>
        <DirectoryServiceTextGd
          className='directory-service-header'
          to={art.routeName}
          $matchpath={!!pathname.match(RegExp(`${art.routeName}$`))}
        >
          graphic
          <br />
          designer
        </DirectoryServiceTextGd>
        <DirectoryServiceTextPg
          className='directory-service-header'
          to={photo.routeName}
          $matchpath={!!pathname.match(RegExp(`${photo.routeName}$`))}
        >
          photo
          <br />
          grapher
        </DirectoryServiceTextPg>
      </DirectoryService>
    </LeftServiceContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  art: selectArtService,
  photo: selectPhotoService,
  engineer: selectEngineerService,
  dev: selectDevService,
  isSideBarActive: selectIsSideBarActive,
});

export default withRouter(connect(mapStateToProps)(ServiceHeader));
