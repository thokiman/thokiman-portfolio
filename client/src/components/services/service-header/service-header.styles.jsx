import { Link } from 'react-router-dom';
import styled, { css, keyframes } from 'styled-components';
const maxWidthTablet = 1024;
const maxWidthDesktop = 1366;
const backgroundColorLeftServiceContainer = 'rgba(18, 129, 148, 0.5)';
const borderColorDirectoryService = 'rgba(18, 129, 148, 0.5)';
const grayColor = '#3f4249';
const whiteSmokeColor = '#e3e1e4';
const lightBlueColor = '#74c6d4';
const maxWidthSmartphone = 800;

const ServiceHeaderIsNotActivated = css`
  display: none;
`;
const getSideBarActive = ({ $issidebaractive }) => {
  if ($issidebaractive) return ServiceHeaderIsNotActivated;
  return null;
};

export const LeftServiceContainer = styled.div`
  width: 25vw;
  height: 68vh;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: ${backgroundColorLeftServiceContainer};
  position: relative;
  left: 4vw;
  transition: height 0.5s ease-in-out, top 0.2s ease;
  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    width: 100vw;
    left: 0;
    ${getSideBarActive}
  }
`;
export const DirectoryService = styled.div`
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-template-rows: repeat(3, 1fr);
  gap: 5px 5px;
  grid-template-areas:
    'directory-service-text-eng directory-service-text-eng directory-service-text-fsd directory-service-text-fsd'
    'directory-service-text-gd directory-service-text-gd directory-service-text-fsd directory-service-text-fsd'
    'directory-service-text-gd directory-service-text-gd directory-service-text-pg directory-service-text-pg';
  height: 76%;
  width: 95%;
  cursor: pointer;
  @media only screen and (max-width: ${maxWidthDesktop}px) {
    width: 98vw;
    height: 20vh;
  }
  @media only screen and (max-width: ${maxWidthTablet}px) {
    width: 98vw;
    height: 15vh;
  }
  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    width: 98vw;
    height: 50vh;
  }
`;
const directoryServiceHover = keyframes`
    0% {
        background-position: 100% 50%;
        color: ${whiteSmokeColor};
    }

    100% {
        background-position: 0% 50%;
        color: ${lightBlueColor};
    } 
`;
const directoryServiceActive = keyframes`
    0% {
        letter-spacing: -0.5em;
        opacity: 0;
    }

    40% {
        opacity: 0.6;
    }

    100% {
        opacity: 1;
    }

`;
const DirectoryServiceTextStyles = css`
  text-decoration: none;
  font-size: 1.9vw;
  height: 100%;
  display: flex;
  align-items: center;
  letter-spacing: 0.02em;
  padding: 0 10px;
  color: ${whiteSmokeColor};
  transition: color 1s linear;
  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    font-size: 6vw;
    padding: 25px;
  }
`;

const DirectoryServiceTextEngStyles = css`
  ${DirectoryServiceTextStyles}
  grid-area: directory-service-text-eng;
  text-align: justify;
  justify-content: flex-end;
  ${
    '' /* @media only screen and (max-width: ${maxWidthDesktop}px) {
    border: red solid 1px;
    height: 10vh;
  }
  @media only screen and (max-width: ${maxWidthTablet}px) {
    border: red solid 1px;
    height: 9vh;
  } */
  }
`;
const DirectoryServiceTextEngHide = css`
  ${DirectoryServiceTextEngStyles}
  border-bottom: ${borderColorDirectoryService} solid 2px;
  border-right: ${borderColorDirectoryService} solid 2px;

  &:hover {
    animation: ${directoryServiceHover} 1s alternate-reverse infinite;
  }
`;
const DirectoryServiceTextEngActive = css`
  ${DirectoryServiceTextEngStyles}
  color: ${grayColor};
  animation: ${directoryServiceActive} 0.7s cubic-bezier(0.215, 0.61, 0.355, 1)
    both;
  cursor: default;
`;

const getDirectoryServiceTextEng = (props) => {
  if (props.$matchpath) return DirectoryServiceTextEngActive;
  return DirectoryServiceTextEngHide;
};
export const DirectoryServiceTextEng = styled(Link)`
  ${getDirectoryServiceTextEng}
`;

const DirectoryServiceTextFsdStyles = css`
  ${DirectoryServiceTextStyles}
  grid-area: directory-service-text-fsd;
  line-height: 1.1em;
  text-align: left;
  justify-content: flex-start;
  ${
    '' /* @media only screen and (max-width: ${maxWidthDesktop}px) {
    border: red solid 1px;
    height: 10vh;
  }
  @media only screen and (max-width: ${maxWidthTablet}px) {
    border: red solid 1px;
    height: 9vh;
  } */
  }
`;
const DirectoryServiceTextFsdHide = css`
  ${DirectoryServiceTextFsdStyles}
  border-bottom: ${borderColorDirectoryService} solid 2px;
  border-left: ${borderColorDirectoryService} solid 2px;

  &:hover {
    animation: ${directoryServiceHover} 1s alternate-reverse infinite;
  }
`;
const DirectoryServiceTextFsdActive = css`
  ${DirectoryServiceTextFsdStyles}
  color: ${grayColor};
  animation: ${directoryServiceActive} 0.7s cubic-bezier(0.215, 0.61, 0.355, 1)
    both;
  cursor: default;
`;
const getDirectoryServiceTextFsd = (props) => {
  if (props.$matchpath) return DirectoryServiceTextFsdActive;
  return DirectoryServiceTextFsdHide;
};
export const DirectoryServiceTextFsd = styled(Link)`
  ${getDirectoryServiceTextFsd}
`;

const DirectoryServiceTextGdStyles = css`
  ${DirectoryServiceTextStyles}
  grid-area: directory-service-text-gd;
  line-height: 1.45em;
  text-align: end;
  justify-content: flex-end;
  ${
    '' /* @media only screen and (max-width: ${maxWidthDesktop}px) {
    border: red solid 1px;
    height: 10vh;
  }
  @media only screen and (max-width: ${maxWidthTablet}px) {
    border: red solid 1px;
    height: 9vh;
  } */
  }
`;
const DirectoryServiceTextGdHide = css`
  ${DirectoryServiceTextGdStyles}
  border-top: ${borderColorDirectoryService} solid 2px;
  border-right: ${borderColorDirectoryService} solid 2px;

  &:hover {
    animation: ${directoryServiceHover} 1s alternate-reverse infinite;
  }
`;
const DirectoryServiceTextGdActive = css`
  ${DirectoryServiceTextGdStyles}
  color: ${grayColor};
  animation: ${directoryServiceActive} 0.7s cubic-bezier(0.215, 0.61, 0.355, 1)
    both;
  cursor: default;
`;
const getDirectoryServiceTextGd = (props) => {
  if (props.$matchpath) return DirectoryServiceTextGdActive;
  return DirectoryServiceTextGdHide;
};
export const DirectoryServiceTextGd = styled(Link)`
  ${getDirectoryServiceTextGd}
`;

const DirectoryServiceTextPgStyles = css`
  ${DirectoryServiceTextStyles}
  grid-area: directory-service-text-pg;
  line-height: 1.1em;
  text-align: left;
  justify-content: flex-start;
  ${
    '' /* @media only screen and (max-width: ${maxWidthDesktop}px) {
    border: red solid 1px;
    height: 10vh;
  }
  @media only screen and (max-width: ${maxWidthTablet}px) {
    border: red solid 1px;
    height: 9vh;
  } */
  }
`;
const DirectoryServiceTextPgHide = css`
  ${DirectoryServiceTextPgStyles}
  border-top: ${borderColorDirectoryService} solid 2px;
  border-left: ${borderColorDirectoryService} solid 2px;

  &:hover {
    animation: ${directoryServiceHover} 1s alternate-reverse infinite;
  }
`;

const DirectoryServiceTextPgActive = css`
  ${DirectoryServiceTextPgStyles}
  color: ${grayColor};
  animation: ${directoryServiceActive} 0.7s cubic-bezier(0.215, 0.61, 0.355, 1)
    both;
  cursor: default;
`;
const getDirectoryServiceTextPg = (props) => {
  if (props.$matchpath) return DirectoryServiceTextPgActive;
  return DirectoryServiceTextPgHide;
};
export const DirectoryServiceTextPg = styled(Link)`
  ${getDirectoryServiceTextPg}
`;

LeftServiceContainer.displayName = 'LeftServiceContainer';
DirectoryService.displayName = 'DirectoryService';
DirectoryServiceTextEng.displayName = 'DirectoryServiceTextEng';
DirectoryServiceTextFsd.displayName = 'DirectoryServiceTextFsd';
DirectoryServiceTextGd.displayName = 'DirectoryServiceTextGd';
DirectoryServiceTextPg.displayName = 'DirectoryServiceTextPg';
