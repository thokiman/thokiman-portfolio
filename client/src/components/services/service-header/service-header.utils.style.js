import {
  hasScreenSize1,
  hasScreenSize1_1,
  hasScreenSize1_2,
  hasScreenSize1_3,
  hasScreenSize2,
  hasScreenSize3_1,
  hasScreenSize3_2,
  hasScreenSize3_3,
  hasScreenSize3_4,
  hasScreenSize3_5,
  hasScreenSize4,
  hasScreenSize5,
  hasScreenSizeWindow,
  hasScreenSizeWindow_1,
} from '../../../utils/common.component.utils';

export const measureDirectoryServiceProps = (viewWidth, viewHeight) => {
  const directoryServiceWidthOffset = 0;
  const directoryServiceHeightOffset = 0;
  let directoryServiceProps;

  if (hasScreenSizeWindow(viewWidth, viewHeight)) {
    console.log('directory-container-service', 1);
    directoryServiceProps = {
      width: 25,
      height: 68,
    };
  } else if (hasScreenSizeWindow_1(viewWidth, viewHeight)) {
    console.log('directory-container-service', 2);
    directoryServiceProps = {
      width: 25 + directoryServiceWidthOffset,
      height: 68 + directoryServiceHeightOffset,
    };
  } else if (hasScreenSize1_1(viewWidth, viewHeight)) {
    console.log('directory-container-service', 3);
    directoryServiceProps = {
      width: 25 + directoryServiceWidthOffset,
      height: 68 + directoryServiceHeightOffset,
    };
  } else if (hasScreenSize1_2(viewWidth, viewHeight)) {
    console.log('directory-container-service', 4);
    directoryServiceProps = {
      width: 25 + directoryServiceWidthOffset,
      height: 68 + directoryServiceHeightOffset,
    };
  } else if (hasScreenSize1_3(viewWidth, viewHeight)) {
    console.log('directory-container-service', 5);
    directoryServiceProps = {
      width: 25 + directoryServiceWidthOffset,
      height: 68 + directoryServiceHeightOffset,
    };
  } else if (hasScreenSize1(viewWidth)) {
    console.log('directory-container-service', 6);
    directoryServiceProps = {
      width: 98 + directoryServiceWidthOffset,
      height: 50 + directoryServiceHeightOffset,
    };
  } else if (hasScreenSize2(viewWidth)) {
    console.log('directory-container-service', 7);
    directoryServiceProps = {
      width: 98 + directoryServiceWidthOffset,
      height: 50 + directoryServiceHeightOffset,
    };
  } else if (hasScreenSize3_1(viewWidth, viewHeight)) {
    console.log('directory-container-service', 8);
    directoryServiceProps = {
      width: 98 + directoryServiceWidthOffset,
      height: 50 + directoryServiceHeightOffset,
    };
  } else if (hasScreenSize3_2(viewWidth, viewHeight)) {
    console.log('directory-container-service', 9);
    directoryServiceProps = {
      width: 98 + directoryServiceWidthOffset,
      height: 50 + directoryServiceHeightOffset,
    };
  } else if (hasScreenSize3_3(viewWidth, viewHeight)) {
    console.log('directory-container-service', 10);
    directoryServiceProps = {
      width: 98 + directoryServiceWidthOffset,
      height: 50 + directoryServiceHeightOffset,
    };
  } else if (hasScreenSize3_4(viewWidth, viewHeight)) {
    console.log('directory-container-service', 11);
    directoryServiceProps = {
      width: 98 + directoryServiceWidthOffset,
      height: 50 + directoryServiceHeightOffset,
    };
  } else if (hasScreenSize3_5(viewWidth, viewHeight)) {
    console.log('directory-container-service', 12);
    directoryServiceProps = {
      width: 98 + directoryServiceWidthOffset,
      height: 50 + directoryServiceHeightOffset,
    };
  } else if (hasScreenSize4(viewWidth, viewHeight)) {
    console.log('directory-container-service', 13);
    directoryServiceProps = {
      width: 98 + directoryServiceWidthOffset,
      height: 50 + directoryServiceHeightOffset,
    };
  } else if (hasScreenSize5(viewWidth)) {
    console.log('directory-container-service', 14);
    directoryServiceProps = {
      width: 98 + directoryServiceWidthOffset,
      height: 50 + directoryServiceHeightOffset,
    };
  } else {
    console.log('directory-container-service', 15);
    directoryServiceProps = {
      width: 98,
      height: 25,
    };
  }

  return directoryServiceProps;
};

export const measureLeftServiceContainerProps = (viewWidth, viewHeight) => {
  const leftServiceContainerWidthOffset = 0;
  const leftServiceContainerHeightOffset = 0;
  let directoryServiceProps;

  if (hasScreenSizeWindow(viewWidth, viewHeight)) {
    console.log('left-service-container-service', 1);
    directoryServiceProps = {
      width: 25 + leftServiceContainerWidthOffset,
      height: 68 + leftServiceContainerHeightOffset,
    };
  } else if (hasScreenSizeWindow_1(viewWidth, viewHeight)) {
    console.log('left-service-container-service', 2);
    directoryServiceProps = {
      width: 25 + leftServiceContainerWidthOffset,
      height: 68 + leftServiceContainerHeightOffset,
    };
  } else if (hasScreenSize1_1(viewWidth, viewHeight)) {
    console.log('left-service-container-service', 3);
    directoryServiceProps = {
      width: 25 + leftServiceContainerWidthOffset,
      height: 68 + leftServiceContainerHeightOffset,
    };
  } else if (hasScreenSize1_2(viewWidth, viewHeight)) {
    console.log('left-service-container-service', 4);
    directoryServiceProps = {
      width: 25 + leftServiceContainerWidthOffset,
      height: 68 + leftServiceContainerHeightOffset,
    };
  } else if (hasScreenSize1_3(viewWidth, viewHeight)) {
    console.log('left-service-container-service', 5);
    directoryServiceProps = {
      width: 25 + leftServiceContainerWidthOffset,
      height: 68 + leftServiceContainerHeightOffset,
    };
  } else if (hasScreenSize1(viewWidth)) {
    console.log('left-service-container-service', 6);
    directoryServiceProps = {
      width: 100 + leftServiceContainerWidthOffset,
      height: 75 + leftServiceContainerHeightOffset,
    };
  } else if (hasScreenSize2(viewWidth)) {
    console.log('left-service-container-service', 7);
    directoryServiceProps = {
      width: 100 + leftServiceContainerWidthOffset,
      height: 80 + leftServiceContainerHeightOffset,
    };
  } else if (hasScreenSize3_1(viewWidth, viewHeight)) {
    console.log('left-service-container-service', 8);
    directoryServiceProps = {
      width: 100 + leftServiceContainerWidthOffset,
      height: 52 + leftServiceContainerHeightOffset,
    };
  } else if (hasScreenSize3_2(viewWidth, viewHeight)) {
    console.log('left-service-container-service', 9);
    directoryServiceProps = {
      width: 100 + leftServiceContainerWidthOffset,
      height: 52 + leftServiceContainerHeightOffset,
    };
  } else if (hasScreenSize3_3(viewWidth, viewHeight)) {
    console.log('left-service-container-service', 10);
    directoryServiceProps = {
      width: 100 + leftServiceContainerWidthOffset,
      height: 60 + leftServiceContainerHeightOffset,
    };
  } else if (hasScreenSize3_4(viewWidth, viewHeight)) {
    console.log('left-service-container-service', 11);
    directoryServiceProps = {
      width: 100 + leftServiceContainerWidthOffset,
      height: 60 + leftServiceContainerHeightOffset,
    };
  } else if (hasScreenSize3_5(viewWidth, viewHeight)) {
    console.log('left-service-container-service', 12);
    directoryServiceProps = {
      width: 100 + leftServiceContainerWidthOffset,
      height: 100 + leftServiceContainerHeightOffset,
    };
  } else if (hasScreenSize4(viewWidth, viewHeight)) {
    console.log('left-service-container-service', 13);
    directoryServiceProps = {
      width: 100 + leftServiceContainerWidthOffset,
      height: 70 + leftServiceContainerHeightOffset,
    };
  } else if (hasScreenSize5(viewWidth)) {
    console.log('left-service-container-service', 14);
    directoryServiceProps = {
      width: 100 + leftServiceContainerWidthOffset,
      height: 50 + leftServiceContainerHeightOffset,
    };
  } else {
    console.log('left-service-container-service', 15);
    directoryServiceProps = {
      width: 100,
      height: 100,
    };
  }

  return directoryServiceProps;
};
