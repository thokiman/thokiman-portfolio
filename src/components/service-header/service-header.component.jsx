import React from "react";
import { Link, withRouter } from "react-router-dom";

import "./service-header.styles.scss";

const ServiceHeader = ({ match: { url }, location: { pathname } }) => {
  return (
    <div className="left-service-container">
      <div className="directory-service">
        <Link
          to={`${url}/engineer`}
          className={
            pathname.match(RegExp(`${url}/engineer$`))
              ? "directory-service-text-eng-active"
              : "directory-service-text-eng"
          }
        >
          engineer
        </Link>
        <Link
          to={`${url}/fullstack`}
          className={
            pathname.match(RegExp(`${url}/fullstack$`))
              ? "directory-service-text-fsd-active"
              : "directory-service-text-fsd"
          }
        >
          full
          <br />
          stack <br />
          developer
        </Link>
        <Link
          to={`${url}/visualart`}
          className={
            pathname.match(RegExp(`${url}/visualart$`))
              ? "directory-service-text-gd-active"
              : "directory-service-text-gd"
          }
        >
          graphic
          <br />
          design
        </Link>
        <Link
          to={`${url}/photo`}
          className={
            pathname.match(RegExp(`${url}/photo$`))
              ? "directory-service-text-pg-active"
              : "directory-service-text-pg"
          }
        >
          photo
          <br />
          graphy
        </Link>
      </div>
    </div>
  );
};

export default withRouter(ServiceHeader);
