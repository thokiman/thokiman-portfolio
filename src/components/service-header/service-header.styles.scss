.left-service-container {
    width: 25%;
    height: 98%;
    display: flex;
    align-items: center;
    justify-content: center;
    text-align: center;
    background-color: rgba($color: #128194, $alpha: .5);
    position: relative;
    left: 40px;
    transition: height 0.5s ease-in-out, top 0.2s ease;



    .directory-service {

        display: grid;
        grid-template-columns: repeat(4, 1fr);
        grid-template-rows: repeat(3, 1fr);
        gap: 5px 5px;
        grid-template-areas:
            "directory-service-text-eng directory-service-text-eng directory-service-text-fsd directory-service-text-fsd"
            "directory-service-text-gd directory-service-text-gd directory-service-text-fsd directory-service-text-fsd"
            "directory-service-text-gd directory-service-text-gd directory-service-text-pg directory-service-text-pg";
        height: 76%;
        width: 95%;
        cursor: pointer;

        .directory-service-text-eng {

            grid-area: directory-service-text-eng;
            text-decoration: none;
            font-size: 235%;
            height: 100%;
            display: flex;
            align-items: center;
            justify-content: center;
            text-align: justify;
            letter-spacing: 0.02em;
            padding: 0 10px;

            color: #e3e1e4;


            border-bottom: rgba($color: #128194, $alpha: .5) solid 2px;
            border-right: rgba($color: #128194, $alpha: .5) solid 2px;
            transition: color 1s linear;


            &:hover {
                animation: directory-service-hover 1s alternate-reverse infinite;
            }

        }

        .directory-service-text-eng-active {

            grid-area: directory-service-text-eng;
            text-decoration: none;
            font-size: 235%;
            height: 100%;
            display: flex;
            align-items: center;
            justify-content: center;
            text-align: justify;
            letter-spacing: 0.02em;
            padding: 0 10px;
            color: #3f4249;


            animation: tracking-in-expand 0.7s cubic-bezier(0.215, 0.610, 0.355, 1.000) both;
            cursor: default;
        }

        .directory-service-text-fsd {

            grid-area: directory-service-text-fsd;
            text-decoration: none;
            font-size: 235%;
            height: 100%;
            display: flex;
            align-items: center;
            justify-content: center;
            text-align: left;
            line-height: 1.1em;
            letter-spacing: 0.02em;
            padding: 0 10px;

            color: #e3e1e4;

            border-bottom: rgba($color: #128194, $alpha: .5) solid 2px;
            border-left: rgba($color: #128194, $alpha: .5) solid 2px;
            transition: color 1s linear;


            &:hover {
                animation: directory-service-hover 1s alternate-reverse infinite;
            }

        }

        .directory-service-text-fsd-active {

            grid-area: directory-service-text-fsd;
            text-decoration: none;
            font-size: 235%;
            height: 100%;
            display: flex;
            align-items: center;
            justify-content: center;
            text-align: left;
            line-height: 1.1em;
            letter-spacing: 0.02em;
            padding: 0 10px;

            color: #3f4249;


            animation: tracking-in-expand 0.7s cubic-bezier(0.215, 0.610, 0.355, 1.000) both;
            cursor: default;
        }

        .directory-service-text-gd {

            grid-area: directory-service-text-gd;
            text-decoration: none;
            font-size: 235%;
            height: 100%;
            display: flex;
            align-items: center;
            justify-content: center;
            text-align: end;
            line-height: 1.1em;
            letter-spacing: 0.02em;
            padding: 0 10px;

            color: #e3e1e4;

            border-top: rgba($color: #128194, $alpha: .5) solid 2px;
            border-right: rgba($color: #128194, $alpha: .5) solid 2px;

            transition: color 1s linear;


            &:hover {
                animation: directory-service-hover 1s alternate-reverse infinite;
            }

        }

        .directory-service-text-gd-active {

            grid-area: directory-service-text-gd;
            text-decoration: none;
            font-size: 235%;
            height: 100%;
            display: flex;
            align-items: center;
            justify-content: center;
            text-align: end;
            line-height: 1.1em;
            letter-spacing: 0.02em;
            padding: 0 10px;

            color: #3f4249;



            animation: tracking-in-expand 0.7s cubic-bezier(0.215, 0.610, 0.355, 1.000) both;
            cursor: default;
        }

        .directory-service-text-pg {

            grid-area: directory-service-text-pg;
            text-decoration: none;
            font-size: 235%;
            height: 100%;
            display: flex;
            align-items: center;
            justify-content: center;
            text-align: left;
            line-height: 1.1em;
            letter-spacing: 0.02em;
            padding: 0 10px;
            padding-right: 55px;
            color: #e3e1e4;


            border-top: rgba($color: #128194, $alpha: .5) solid 2px;
            border-left: rgba($color: #128194, $alpha: .5) solid 2px;
            transition: color 1s linear;


            &:hover {
                animation: directory-service-hover 1s alternate-reverse infinite;
            }

        }

        .directory-service-text-pg-active {

            grid-area: directory-service-text-pg;
            text-decoration: none;
            font-size: 235%;
            height: 100%;
            display: flex;
            align-items: center;
            justify-content: center;
            text-align: left;
            line-height: 1.1em;
            letter-spacing: 0.02em;
            padding: 0 10px;
            padding-right: 55px;
            color: #3f4249;




            animation: tracking-in-expand 0.7s cubic-bezier(0.215, 0.610, 0.355, 1.000) both;
            cursor: default;
        }
    }
}

@keyframes directory-service-hover {
    0% {
        background-position: 100% 50%;
        color: #e3e1e4;
    }

    100% {
        background-position: 0% 50%;
        color: #74c6d4;
    }
}

@keyframes tracking-in-expand {
    0% {
        letter-spacing: -0.5em;
        opacity: 0;
    }

    40% {
        opacity: 0.6;
    }

    100% {
        opacity: 1;
    }
}