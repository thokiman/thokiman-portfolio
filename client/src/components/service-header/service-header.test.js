import React from "react";
import { shallow } from "enzyme";

import { ServiceHeader } from "./service-header.component";

describe("ServiceHeader component", () => {
  let wrapper;

  const mockFalseBool = false;
  const mockRoute = "/";
  beforeEach(() => {
    const mockProps = {
      location: { pathname: mockRoute },
      art: { routeName: mockRoute },
      photo: { routeName: mockRoute },
      engineer: { routeName: mockRoute },
      fullstack: { routeName: mockRoute },
      isSideBarHidden: mockFalseBool,
    };
    wrapper = shallow(<ServiceHeader {...mockProps} />);
  });

  it("should render ServiceHeader component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
