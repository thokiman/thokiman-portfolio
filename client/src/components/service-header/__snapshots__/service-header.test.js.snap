// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceHeader component should render ServiceHeader component 1`] = `ShallowWrapper {}`;
